src/implementation/MassaWalletProviders.ts

```diff
@@ -0,0 +1,120 @@
+import { uid } from 'uid';
+import { ICustomEventMessageResponse } from '../interfaces/ICustomEventMessageResponse';
+import { ICustomEventMessageRequest } from '../interfaces/ICustomEventMessageRequest';
+import { AvailableCommands } from './Commands';
+import { IRegisterEvent } from '../interfaces';
+
+const MASSA_WINDOW_OBJECT = 'massaWalletProvider';
+
+type CallbackFunctionVariadicAnyReturn = (
+  error: Error | null,
+  result: any,
+) => any;
+
+// =========================================================
+
+export class MassaWalletProviders {
+  private static instance: MassaWalletProviders;
+  private registeredProviders: { [key: string]: string } = {};
+  private pendingRequests: Map<string, CallbackFunctionVariadicAnyReturn>;
+
+  public static init(): MassaWalletProviders {
+    if (!MassaWalletProviders.instance) {
+      MassaWalletProviders.instance = new MassaWalletProviders();
+    }
+
+    return MassaWalletProviders.instance;
+  }
+
+  private constructor() {
+    this.getWalletProviders = this.getWalletProviders.bind(this);
+    this.handleResponseFromContentScript =
+      this.handleResponseFromContentScript.bind(this);
+    this.sendMessageToContentScript =
+      this.sendMessageToContentScript.bind(this);
```

Comment:
What is this ?

Reply:
these are binding each of the funcs to these , but I have removed it as its not necessary anymore.